pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1272,3 +1277,11 @@ func setTerminationReasonByPhase(phase v1alpha1.MachinePhase, terminationConditi
 		terminationCondition.Message = "Machine Controller is scaling down machine"
 	}
 }
+
+func getProviderID(machine *v1alpha1.Machine) string {
+	return machine.Spec.ProviderID
+}
+
+func getNodeName(machine *v1alpha1.Machine) string {
+	return machine.Status.Node
+}
```

Comment:
I have changed the definition as the earlier definition where I was looking for nodeName is MachineSpec was not printing anything.

Reply:
Also there is no panic situation as we discussed but empty nodeName might be printed

Reply:
Okay thanks for this.